public/app/features/dashboard/dashgrid/PanelPerformanceMonitor.tsx

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useRef } from 'react';
+
+import { FieldConfigSource } from '@grafana/data';
+import { faro } from '@grafana/faro-web-sdk';
+import appEvents from 'app/core/app_events';
+import { DashboardSavedEvent } from 'app/types/events';
+
+interface Props {
+  children: JSX.Element;
+  isInPanelEdit: boolean;
+  panelType: string;
+  panelId: number;
+  panelTitle: string;
+  panelOptions: any;
+  panelFieldConfig: FieldConfigSource;
+}
+
+export const PanelPerformanceMonitor = (props: Props) => {
+  const startLoadTime = performance.now();
+  const panelIdStr = String(props.panelId);
+
+  const oldPanelOptions = useRef(props.panelOptions);
+  const oldPanelFieldConfig = useRef<FieldConfigSource>(props.panelFieldConfig);
+
+  const savedFieldConfig = useRef<FieldConfigSource>(props.panelFieldConfig);
+  const savedPanelOptions = useRef(props.panelOptions);
+
+  const logChanges = () => {
+    logPanelOptionChanges(savedPanelOptions.current, oldPanelOptions.current);
+    logFieldConfigChanges(savedFieldConfig.current, oldPanelFieldConfig.current);
+
+    //set the old values to the current values for next save
+    oldPanelOptions.current = savedPanelOptions.current;
+    oldPanelFieldConfig.current = savedFieldConfig.current;
+  };
+
+  useEffect(() => {
+    savedPanelOptions.current = props.panelOptions;
+  }, [props.panelOptions]);
+
+  useEffect(() => {
+    savedFieldConfig.current = props.panelFieldConfig;
+  }, [props.panelFieldConfig]);
+
+  const logEvent = (eventName: string, newKey: string, newVal: string, oldVal?: string) => {
+    const logObj = {
+      key: newKey,
+      newValue: newVal,
+      oldValue: oldVal ?? '',
+      panelTitle: props.panelTitle,
+      panelId: panelIdStr,
+      panelType: props.panelType,
+    };
+
+    //TODO: do batch push when that becomes available
```

Comment:
is there a github issue we can track this at? may make sense to include that here for context / future us

Reply:
I understood when using Faro API, it uses the [EchoSrvTransport](https://github.com/grafana/grafana/blob/main/public/app/core/services/echo/backends/grafana-javascript-agent/EchoSrvTransport.ts) which makes the Faro reports through EchoSrv. And EchoSrv by default batches events for 10s. So, what batching are we referring to?

Reply:
@ivanortegaalba you're right! Thanks! Batching works as expected now. I used an older faro-web-sdk version when I started working on this, then updated to the latest version but remained under the impression that it's still wip. Will remove the comment